### DISCUSSION TITLE
the correspond between detpt.ppath   and traj.data id

### DISCUSSION THREAD

**Discussion by 1234soda** [fangq/mcx#229] [Help]
I want to ask whether the data in row n of dept.ppath corresponds to the total path of photon id =(n-1) traveling in the medium? Does it correspond to the first column id sequence in traj.data?

**Comment by 1234soda** [comment]
I also want to know how dees dept.ppath is calculated? Does each row of ppath correspond to the total flight path of an individual photon through the medium?
Thanks for your reply!

**Comment by fangq** [comment]
hi @1234soda, the n-th row of detp.ppath stores the n-th detected photon packet. it is not the linear index of the simulated photon.

however, if you use cfg.bc='______111111' to capture all photon packets, the index may follow the simulation photon index, but if any photon is terminated due to exceeding the maximum time gate, it will not be captured and will create a jump (shift in the index) - so it is not reliable.

> I also want to know how dees dept.ppath is calculated? Does each row of ppath correspond to the total flight path of an individual photon through the medium?

correct.

**Reply by 1234soda** [reply]
> hi @1234soda, the n-th row of detp.ppath stores the n-th detected photon packet. it is not the linear index of the simulated photon.
> 
> however, if you use cfg.bc='______111111' to capture all photon packets, the index may follow the simulation photon index, but if any photon is terminated due to exceeding the maximum time gate, it will not be captured and will create a jump (shift in the index) - so it is not reliable.
> 
> > I also want to know how dees dept.ppath is calculated? Does each row of ppath correspond to the total flight path of an individual photon through the medium?
> 
> correct.

OK, because recently I was counting the average flight time of photons to the lowest depth, and found that I drew the three-dimensional flight path of some photons. The path clearly looked very long, but its corresponding detp.ppath was only a few mm, so the above question arose. It seems that there is something wrong with my statistics, thank you very much for your reply!
